tests/main/test_integration.py

```diff
@@ -687,6 +738,7 @@ def test_full_datapath(
         decimal=4,
     )
 
+    # Tanner (12/29/20): Good to do this at the end of tests to make sure they don't cause problems with other integration tests
     test_process_manager.hard_stop_and_join_processes()
     assert is_queue_eventually_empty(da_error_queue) is True
```

Comment:
doesn't the error queue get drained as part of the `hard_stop` itself?  so asserting that it is empty after the hard stop would be always true even if there were an error? would it be better to check the dictionary returned by hard_stop to confirm there's nothing in the spot where error_queue items would be put for data_analyzer?

Reply:
Yes this would be better